README
Debian Packaging Tutorial
=========================

Contact: Lucas Nussbaum <lucas@debian.org>

This tutorial is an introduction to Debian packaging. It teaches prospective
developers how to modify existing packages, how to create their own packages,
and how to interact with the Debian community.

It includes a tutorial and three practical sessions:
 - modifying the 'grep' package
 - packaging the 'gnujump' game from scratch
 - packaging a Java library

Contributing to this tutorial
-----------------------------
This tutorial is available on git.debian.org:
Gitweb: http://git.debian.org/?p=collab-maint/packaging-tutorial.git
or git clone git://git.debian.org/collab-maint/packaging-tutorial.git

Thanks
------
Many people have contributed to this tutorial, and it would not be
possible to list them all there. In particular:
- Phil Hands contributed to the slides while preparing a tutorial for UKUUG;
  see http://www.ukuug.org/events/debian2011/

Legal stuff
-----------
This tutorial is
Copyright (c) 2011 Lucas Nussbaum <lucas@debian.org>

This document is free software: you can redistribute it and/or modify it under
either (at your option):
- The terms of the GNU General Public License as published by the Free Software
  Foundation, either version 3 of the License, or (at your option) any later
  version.
  On Debian systems, the complete text of the GNU General
  Public License version 3 can be found in `/usr/share/common-licenses/GPL-3'.
  It can also be found on the web: http://www.gnu.org/licenses/gpl.html
- The terms of the Creative Commons Attribution-ShareAlike 3.0 Unported License,
  which can be found at http://creativecommons.org/licenses/by-sa/3.0/legalcode.


debian/changelog
packaging-tutorial (0.15) unstable; urgency=medium

  [ Helge Kreutzmann ]
  * Unfuzzy German translation.

  [ Lucas Nussbaum ]
  * Unfuzzy French translation.
  * Update standards-version. No changes needed.

 -- Lucas Nussbaum <lucas@debian.org>  Thu, 16 Oct 2014 20:15:36 +0200

packaging-tutorial (0.14) unstable; urgency=low

  [ Helge Kreutzmann ]
  * Unfuzzy German translation.

  [ Lucas Nussbaum ]
  * Add a practical session for packaging a CPAN module.
    Patch from Gregor Herrmann <gregoa@debian.org>. Thanks!
    Closes: #759758
  * Update portuguese translation. Thanks to
    Américo Monteiro <a_monteiro@gmx.com> for the patch.
    Closes: #756201

 -- Lucas Nussbaum <lucas@debian.org>  Sun, 31 Aug 2014 21:17:48 -0700

packaging-tutorial (0.13) unstable; urgency=medium

  * Add alternative dependency on texlive-lang-japanese following
    the split of texlive-lang-cjk. Thanks to Gregor Herrmann for the
    patch! Closes: #751044

 -- Lucas Nussbaum <lucas@debian.org>  Sun, 29 Jun 2014 18:32:42 +0200

packaging-tutorial (0.12) unstable; urgency=low

  * Add Japanese translation from KURASAWA Nozomu (nabetaro)
    nabetaro@caldron.jp>.
  * Bump Standards-Version to 3.9.5. No changes needed.
  * Fix typo in ru's doc-base file.

 -- Lucas Nussbaum <lucas@debian.org>  Thu, 31 Oct 2013 09:56:38 +0100

packaging-tutorial (0.11) unstable; urgency=low

  * Add Russian translation from Lev Lamberov <l.lamberov@gmail.com>.
  * Add build-depend on texlive-lang-cyrillic.
  * Russian translation update after review from debian-l10n-russian.
  * Add doc-base file for Russian.
  * Add Portuguese translation. Closes: #724782
  * Minor fixes in portuguese translation to make it build.
  * Add a README for translators.

 -- Lucas Nussbaum <lucas@debian.org>  Tue, 29 Oct 2013 20:40:55 +0100

packaging-tutorial (0.10) unstable; urgency=low

  [ Lucas Nussbaum ]
  * Change contact email to packaging-tutorial@p.d.o.
  * Mention ITP bugs.
  * Added documentation about the BTS.
  * Mention localized lists (Closes: #719925).
  * Update version in .tex.
  * Add slide: checklist before asking for sponsorship.
  * Improve 'Contribute' slide.
  * Update .po / .pot files.
  * Update CDBS/dh stats.

  [ Helge Kreutzmann ]
  * Fix two minor translation issues
  * Update German translation

  [ Cédric Boutillier ]
  * update French translation
  * proofread [ Jean-Pierre Giraud ]

 -- Lucas Nussbaum <lucas@debian.org>  Thu, 12 Sep 2013 08:21:00 +0200

packaging-tutorial (0.9) unstable; urgency=low

  [ Helge Kreutzmann ]
  * Unfuzzy strings

  [ Cédric Boutillier ]
  * (fr) update translation [proofread by Baptiste, JP Guillonneau]
  * (fr) proofread [Philippe Piette]

  [ Lucas Nussbaum ]
  * Update translation files
  * Add a debian/rule target to warn translators.
  * Fix typo. Reported by William King.
  * Add build-dependencies on texlive-lang-{german,spanish,french}.
    Closes: #712357
  * Fix minor typo. Patch from Simon Ruderich.
  * More minor fixes from Simon Ruderich.
  * Use fragile=singleslide for a faster build. Patch from Simon Ruderich.
    Closes: #702608
  * Update URLs to devscripts Vcs. Spotted by Niels Thykier. Closes: #704194
  * Manually unfuzzy french, german and spanish translations.

 -- Lucas Nussbaum <lucas@debian.org>  Thu, 20 Jun 2013 18:46:19 +0200

packaging-tutorial (0.8) unstable; urgency=low

  * Update spanish translation. Thanks to Omar Campagne. Closes: #681338
  * Add improvements proposed by Helge Kreutzmann. Closes: #672683
  * Add new mandatory targets in debian/rules. Thanks to Julien Cristau.
    Closes: #689287
  * Fix ES translation to make the package build again.
  * Re-add note to translators.
  * Fix what I could fix in DE, FR, ES translations.
  * Add doc-base entries for .es and .de. Closes: #680180
  * debian/copyright: Document copyright of the Debian logo. Closes: #685904
  * Fix gnujump URL. Thanks Henning Sprang and Stéphane Kanschine.
    Closes: #685359
  * Check for wrong version in .tex during build
  * Clarify that mk-build-deps is useful if package not uploaded yet.
    Closes: #702119
  * Include many fixes from Francesco Poli. Closes: #686594
  * Mention Debian Maintainer Dashboard.
  * Integrate more fixes from Francesco Poli. Closes: #695534

 -- Lucas Nussbaum <lucas@debian.org>  Sun, 03 Mar 2013 11:50:43 +0100

packaging-tutorial (0.7) unstable; urgency=low

  [ Helge Kreutzmann ]
  * Add German translation. Closes: #672682.

  [ Stefano Zacchiroli ]
  * LaTeX markup: turn \url{} into hyperlinks using hyperref
  * LaTeX markup: remove unescaped \textbf{} within lstlisting
  * DEP-5: point to the policy version of the standard

  [ Lucas Nussbaum ]
  * Drop the use of LATEXFILTER to ease possible backports of the package.
    That script was renamed in latex-make 2.1.17-1. Closes: #676748.
  * Mention that fakeroot must be installed as well as devscripts.
    Closes: #659126.
  * es.po: unfuzzied a few strings that did not require language skills.
  * Bump Standards-Version to 3.9.3. No changes needed.

 -- Lucas Nussbaum <lucas@debian.org>  Wed, 04 Jul 2012 11:47:13 +0200

packaging-tutorial (0.6) unstable; urgency=low

  * Explain -us -uc when building packages.
  * Mention dch -i.
  * Explain how to install build-deps using mk-build-deps.
  * Explain how to unpack the source with dpkg-source -x.
  * Improve title of VCS slide.
  * Give example of package adoption.
  * update po files
  * update date in document
  * update po files

 -- Lucas Nussbaum <lucas@debian.org>  Mon, 09 Apr 2012 22:32:06 +0200

packaging-tutorial (0.5) unstable; urgency=low

  [ Lucas Nussbaum ]
  * Minor updates to french translation during preparation for Capitole du
    Libre 2011.
  * Add a debian/rules target to regenerate the version and the date in the
    document.
  * Use newer version of grep with recent debian/ubuntu releases due to the ld
    --as-needed change.
  * Do not mention patch-tracker twice.

update-cdbs-dh7.rb
#!/usr/bin/ruby -w

require 'date'

BROKEN_DATES = ['2011-05-16', '2011-05-17', '2011-05-18', '2011-05-19', '2011-05-20', '2013-05-15', '2013-05-21', '2013-05-22', '2013-05-23', '2013-05-24', '2013-05-25', '2013-05-26', '2013-05-27', '2013-05-28', '2013-05-29', '2013-05-30', '2013-07-03', '2013-07-04', '2013-03-23', '2013-03-24', '2013-03-25', '2012-12-30', '2013-05-14', '2013-05-15' ]

system("scp lintian.debian.org:~cjwatson/dhstats.txt .")
f = File::new("cdbs-dh7.txt", 'w')
f.puts "date dh dh7 cdbs"
IO::read("dhstats.txt").split(/\n\n/).each do |par|
  par.strip!
  next if par == "Sun Sep 20 00:29:01 UTC 2009"
  lines = par.split(/\n/)
  date = Date::parse(lines[0]).to_s
  next if BROKEN_DATES.include?(date) or date > '2013-05-13'
  dh = lines.grep(/^debhelper: /)[0].split[2].gsub(/[\(\)\%]/,'').to_f
  dh7 = lines.grep(/^dh\(1\): /)[0].split[2].gsub(/[\(\)\%]/,'').to_f
  cdbs = lines.grep(/^CDBS: /)[0].split[2].gsub(/[\(\)\%]/,'').to_f
  f.puts "#{date} %.2f %.2f %.2f" % [dh - dh7 - cdbs, dh7, cdbs]
end
f.close
system("rm dhstats.txt")
